assembly/src/ast/parsers/debug.rs

```diff
@@ -0,0 +1,36 @@
+use super::{
+    parse_checked_param,
+    Instruction::*,
+    Node::{self, Instruction},
+    ParsingError, Token,
+};
+use vm_core::DebugOptions;
+
+// INSTRUCTION PARSERS
+// ================================================================================================
+
+/// Returns `Debug` instruction node.
+///
+/// # Errors
+/// Returns an error if the instruction token contains a wrong number of parameters, or if
+/// the provided parameters are not valid.
+pub fn parse_debug(op: &Token) -> Result<Node, ParsingError> {
+    debug_assert_eq!(op.parts()[0], "debug");
+    if op.num_parts() < 2 {
```

Comment:
Maybe we should support a baseline `debug` decorator with no arguments.  This should print out all debug information to the client i.e. comprehensive stack, memory and locals info (at the minute it will only be all stack).

Reply:
I'm a bit hesitant about this as this could print out a lot of info which may be difficult to comprehend. If the user really wants this info, the workaround is to add a couple more instructions - e.g.,
```
begin
  <some instructions>
  debug.stack
  debug.mem
  debug.local
  <some instructions>
end
```

Reply:
I agree, this would be excessive to print. I was thinking more in terms of programatic interrogation of the debug info.  Say we have a debug that takes an identifier argument, which we can then implement a handler to interrogate the data at different points in the program.

```
begin
    ...
    debug.id_x
    ...
    debug.id_y
    ...
end
``` 

with a debug handler defined something like:

```rust
    fn handle_debug(id: Option<String>, clk: u32, stack: Vec<Felt>, n: usize, mem: Vec<Word>, local: Vec<Word>) {
        match id {
             Some("id_x") => {// interrogate vm at state id_x},
             Some("id_y") => {// interrogate vm at state id_y},
             _ => unreachable()
        }
    }
```